src/fifo.rs

```diff
@@ -0,0 +1,180 @@
+    /// FIFO_SRC bit 6 OVRN
+    pub (crate) const OVRN: u8 = 0b0100_0000;
+    /// FIFO_SRC bits 5:0 FSS
+    pub (crate) const FSS: u8 = 0b0011_1111;
+    /// CTRL_REG9 FIFO-related settings 
+    pub (crate) const CTRL_REG9_FIFO: u8 = 0b0001_0011;
+    /// Decimation setting in CTRL_REG5_XL
+    pub (crate) const DEC: u8 = 0b1100_0000;
+}
+
+/// FIFO settings
+#[derive(Debug)]
+pub struct FIFOConfig {    
+    /// FIFO memory enable
+    pub fifo_enable: bool, 
+    /// Select FIFO operation mode (see Table 84 for details)        
+    pub fifo_mode: FIFO_MODE, // default Bypass    
+    /// Enable threshold level use
+    pub fifo_use_threshold: bool, 
+    /// Set the threshold level
+    pub fifo_threshold: u8, // default 32
+    /// Store temperature data in FIFO
+    fifo_temperature_enable: bool,     
+    }
+ 
+impl Default for FIFOConfig {
+    fn default() -> Self {
+        FIFOConfig {                      
+            fifo_enable: false, // disabled
+            fifo_mode: FIFO_MODE::Bypass, // Bypass mode            
+            fifo_use_threshold: false, // FIFO depth not limited
+            fifo_threshold: 32u8, // set the threshold level      
+            fifo_temperature_enable: false, // temperature data not stored in FIFO
+            }
+        }
+    }
+
+impl FIFOConfig {
+    /// Returns values to be written to CTRL_REG9 and FIFO_CTRL:   
+    fn f_fifo_ctrl(&self) -> u8 {
+        let mut data = 0u8;
+        data |= self.fifo_mode.value();
+        data |= self.fifo_threshold;        
+        data
+        }
+    fn f_ctrl_reg9(&self) -> u8 {
+        let mut data = 0u8;
+        if self.fifo_temperature_enable {data |= 1 << 4;}        
+        if self.fifo_enable { data |= 1 << 1;}
+        if self.fifo_use_threshold { data |= 1;}        
+        data
+        }   
+    }
+
+#[derive(Debug)]
+/// Contents of the FIFO_STATUS register (threshold reached, overrun, empty, stored data level)
+pub struct FifoStatus {
+    /// FIFO threshold status. True if FIFO filling is equal or higher than    threshold level
+    pub fifo_thresh_reached: bool,
+    /// True is FIFO is completely filled and at least one samples has been overwritten
+    pub fifo_overrun: bool,
+    /// True if FIFO is empty (level = 0)
+    pub fifo_empty: bool,
+    /// Number of unread samples stored into FIFO
+    pub fifo_level: u8,    
+    }
+ 
+impl<T> LSM9DS1<T>
+where
+    T: Interface,
+    {
+    
+    /// Enable and configure FIFO
+    pub fn configure_fifo(&mut self, config: FIFOConfig) -> Result<(), T::Error> {                
+        
+        // write values to the FIFO_CTRL register
+        self.interface.write(Sensor::Accelerometer, register::AG::FIFO_CTRL.addr(), config.f_fifo_ctrl())?;         
+
+        // write values to specific bits of the CTRL_REG9 register        
+        let ctrl_reg9: u8 = self.read_register(Sensor::Accelerometer, register::AG::CTRL_REG9.addr())?; 
+        //let mask: u8 = 0b00010011;
```

Comment:
I would remove this.

Reply:
Which part, not sure what you meant here?

Reply:
Sorry, it was confusing. I meant line 96, `//let mask: u8 = 0b00010011;`. If we don't use `mask`, it should not be there.

Reply:
Removed, the mask is used but from the `FIFO_Bitmasks` struct.